### DISCUSSION TITLE
Choice of the Elastic license?

### DISCUSSION THREAD

**Discussion by i-baumann** [posit-dev/positron#3820] [Q&A] [answered]
Just curious about the choice of license, especially one not approved by the OSI. AGPLv3 suffices for individual RStudio use while offering that IDE in an enterprise product, so wondering what's different in Positron's case. Should I be concerned that Positron won't be free in the future or is this an artifact of other components of the IDE that aren't open source or can't be made open source?

**Comment by jmcphers** [comment, marked as answer]
Positron will always be free (just like RStudio). Our choice of the Elastic license over AGPL for Positron has a lot to do with our experience with RStudio and the way its license was occasionally abused (obviously can't go into details about that here). If you read the license terms in detail you'll find that despite not being OSI approved (due to the hosting clause), the Elastic license is simpler and grants more freedom than the AGPL in many ways.

We were able to bring some parts of Positron under the MIT license, including a new [high-performance R backend](https://github.com/posit-dev/ark) that can be used with other IDEs. 

We've got more info on our wiki here: https://github.com/posit-dev/positron/wiki/Licensing

**Reply by agifi** [reply]
Can you be a bit more specific when you say that the  'license was occasionally abused'?

Either someone used the freedoms granted under the AGPL in a way you did not like and then I would say: tough luck - this is how open source works.

Alternatively, someone infringed on the license and then you should just enforce the license.

Punishing the community, however, by not going for an open source license is really deplorable, especially for a company that calls itself an 'open source data science' company.

**Reply by juliasilge** [reply]
We understand that licensing can be a contentious topic and people come to it with a variety of perspectives and priorities. If you decide Positron is not for you because of its license, we get it! As we've said elsewhere, this decision is based on Posit's goals for building a business that is sustainable in the long term and allows us to continue to invest in tools for the data science community, including quite a fair amount of software that has OSI-approved licenses.

Because there can be confusion about what these non-OSI-approved licenses mean, I do want reiterate that under the Elastic license, Positron is:

- free to use, including for commercial (plus academic, research, personal, etc) use
- source available

The primary restriction under this license is that you can't host Positron as a service to third parties (without an agreement).

**Reply by vexingly** [reply]
> We understand that licensing can be a contentious topic and people come to it with a variety of perspectives and priorities. If you decide Positron is not for you because of its license, we get it! As we've said elsewhere, this decision is based on Posit's goals for building a business that is sustainable in the long term and allows us to continue to invest in tools for the data science community, including quite a fair amount of software that has OSI-approved licenses.
> 
> Because there can be confusion about what these non-OSI-approved licenses mean, I do want reiterate that under the Elastic license, Positron is:
> 
> * free to use, including for commercial (plus academic, research, personal, etc) use
> * source available
> 
> The primary restriction under this license is that you can't host Positron as a service to third parties (without an agreement).

Thank you for the clarification @juliasilge, it seems that hosting Positron as a service (not to third parties) still requires a license key, who would we be able to discuss this with?

**Reply by juliasilge** [reply]
Thank you for your interest @vexingly! We're not planning on releasing a standalone server product for Positron (i.e. no analog for RStudio Server). Instead there are two options for desktop-to-server modes of use:

- Folks can use remote SSH sessions; an experimental version of this is available now and you can read some details in https://github.com/posit-dev/positron/pull/4251.
- We'll have preview support for Positron on Posit Workbench before the end of the year; you can read a few more details on this in https://github.com/posit-dev/positron/discussions/4936#discussioncomment-10886259.

**Reply by vexingly** [reply]
> Thank you for your interest @vexingly! We're not planning on releasing a standalone server product for Positron (i.e. no analog for RStudio Server). Instead there are two options for desktop-to-server modes of use:
> 
> * Folks can use remote SSH sessions; an experimental version of this is available now and you can read some details in [Experimental support for remote SSH sessions #4251](https://github.com/posit-dev/positron/pull/4251).
> * We'll have preview support for Positron on Posit Workbench before the end of the year; you can read a few more details on this in [Positron-Server (Use positron in Web browser)? #4936 (comment)](https://github.com/posit-dev/positron/discussions/4936#discussioncomment-10886259).

Thank you for the links @juliasilge, those options are not very compatible with our organization so we will have to continue to recommend RStudio Server until there is a replacement for that. We do look forward to watch the evolution of the Positron product and can hope there will be more options in the future.
